sql/src/test/java/io/crate/planner/consumer/CrossJoinConsumerTest.java

```diff
@@ -0,0 +1,242 @@
+        expectedException.expect(ValidationException.class);
+        expectedException.expectMessage("At least 2 relations are required for a CROSS JOIN");
+
+        MultiSourceSelect statement = new MultiSourceSelect(
+                ImmutableMap.<QualifiedName, AnalyzedRelation>of(new QualifiedName("foo"), mock(QueriedTable.class)),
+                ImmutableList.<OutputName>of(),
+                mock(QuerySpec.class)
+        );
+        ConsumerContext context = new ConsumerContext(statement, plannerContext);
+        consumer.consume(statement, context);
+        throw context.validationException();
+    }
+
+    @Test
+    public void testFunctionWithJoinCondition() throws Exception {
+        NestedLoop plan = plan("select u1.name || u2.name from users u1, users u2");
+        TopNProjection topN = (TopNProjection) plan.localMergePhase().projections().get(0);
+        assertThat(topN.outputs().get(0), isFunction("concat"));
+    }
+
+    @Test
+    public void testJoinConditionInWhereClause() throws Exception {
+        NestedLoop plan = plan("select * from users u1, users u2 where u1.name || u2.name = 'foobar'");
+        Projection projection = plan.localMergePhase().projections().get(0);
+        assertThat(projection, instanceOf(FilterProjection.class));
+        Symbol query = ((FilterProjection) projection).query();
+        assertThat(query, isFunction(EqOperator.NAME));
+    }
+
+    @Test
+    public void testExplicitCrossJoinWithoutLimitOrOrderBy() throws Exception {
+         NestedLoop plan = plan("select u1.name, u2.name from users u1 cross join users u2");
+
+         assertThat(plan.localMergePhase().projections().size(), is(1));
+         TopNProjection topN = ((TopNProjection) plan.localMergePhase().projections().get(0));
+
+         assertThat(topN.limit(), is(Constants.DEFAULT_SELECT_LIMIT));
+         assertThat(topN.offset(), is(0));
+         assertThat(topN.outputs().size(), is(2));
+
+
+         assertThat(plan.left().resultNode().projections().size(), is(0));
+         MergePhase leftMerge = plan.nestedLoopPhase().leftMergePhase();
+         assertThat(leftMerge.projections().size(), is(0));
+
+         assertThat(plan.right().resultNode().projections().size(), is(0));
+         MergePhase rightMerge = plan.nestedLoopPhase().rightMergePhase();
+         assertThat(rightMerge.projections().size(), is(0));
+    }
+
+    @SuppressWarnings("ConstantConditions")
+    @Test
+    public void testOrderByPushDown() throws Exception {
+        NestedLoop plan = plan("select u1.name, u2.name from users u1, users u2 order by u1.name");
+        assertThat(plan.left().resultNode().projections().size(), is(1));
+
+        QueryAndFetch leftPlan = (QueryAndFetch) plan.left().plan();
+        CollectPhase collectPhase = leftPlan.collectNode();
+        assertThat(collectPhase.projections().size(), is(1));
+        TopNProjection topN = ((TopNProjection) collectPhase.projections().get(0));
```

Comment:
please change the comment in https://github.com/crate/crate/pull/2231/files#diff-7159fb3ea2b2d07a81f969fbc77c1f7dR210 it's confiusing